website/docs/script/vm-cycle-limits.mdx

```diff
@@ -0,0 +1,163 @@
+
+### Instruction Cycles
+
+All CKB VM instructions consume 1 cycle except the following ones:
+
+| Instruction | Cycles               |
+| ----------- | -------------------- |
+| JALR        | 3                    |
+| JAL         | 3                    |
+| J           | 3                    |
+| JR          | 3                    |
+| BEQ         | 3                    |
+| BNE         | 3                    |
+| BLT         | 3                    |
+| BGE         | 3                    |
+| BLTU        | 3                    |
+| BGEU        | 3                    |
+| BEQZ        | 3                    |
+| BNEZ        | 3                    |
+| LD          | 2                    |
+| SD          | 2                    |
+| LDSP        | 2                    |
+| SDSP        | 2                    |
+| LW          | 3                    |
+| LH          | 3                    |
+| LB          | 3                    |
+| LWU         | 3                    |
+| LHU         | 3                    |
+| LBU         | 3                    |
+| SW          | 3                    |
+| SH          | 3                    |
+| SB          | 3                    |
+| LWSP        | 3                    |
+| SWSP        | 3                    |
+| MUL         | 5                    |
+| MULW        | 5                    |
+| MULH        | 5                    |
+| MULHU       | 5                    |
+| MULHSU      | 5                    |
+| DIV         | 32                   |
+| DIVW        | 32                   |
+| DIVU        | 32                   |
+| DIVUW       | 32                   |
+| REM         | 32                   |
+| REMW        | 32                   |
+| REMU        | 32                   |
+| REMUW       | 32                   |
+| ECALL       | 500 (see note below) |
+| EBREAK      | 500 (see note below) |
+
+### Syscall Cycles
+
+As shown in the above chart, each syscall will have 500 initial cycle consumptions. This is based on real performance metrics gathered benchmarking CKB implementation, certain bookkeeping logics are required for each syscall here.
+
+In addition, for each byte loaded into CKB VM in the syscalls, 0.25 cycles will be charged. Notice fractions will also be rounded up here, so 30.25 cycles will become 31 cycles.
+
+## Guidelines
+
+In general, the cycle consumption rules above follow certain guidelines:
+
+- Branches are more expensive than normal instructions.
+- Memory accesses are more expensive than normal instructions. Since CKB VM is a 64-bit system, loading 64-bit value directly will cost less cycle than loading smaller values.
+- Multiplication and divisions are much more expensive than normal instructions.
+- Syscalls include 2 parts: the bookkeeping part at first, and a plain memcpy phase. The first bookkeeping part includes quite complex logic, which should consume much more cycles. The memcpy part is quite cheap on modern hardware, hence less cycles will be charged.
```

Comment:
```suggestion
- Memory accesses are more expensive than normal instructions. Since CKB-VM is a 64-bit system, loading a 64-bit value directly will cost fewer cycles than loading smaller values.
- Multiplication and divisions are much more expensive than normal instructions.
- Syscalls include 2 parts: the bookkeeping part at first, and a plain memcpy phase. The first bookkeeping part includes quite complex logic, which should consume much more cycles. The memcpy part is quite cheap on modern hardware, hence fewer cycles will be charged.
```